lib/hbs.js

```diff
@@ -69,8 +69,7 @@ function middleware(filename, options, cb) {
           cb(null, res);
         });
       } catch (err) {
-        err.message = filename + ': ' + err.message;
-        cb(err);
+        cb(new Error(filename + ': ' + err.message));
```

Comment:
I looked into this change too, but was hoping for a backwards compatible one so this didn't need to be a 5.0 release. It drops all custom properties from the original Error.

Reply:
Hmm... What kind of custom properties are present here? The stack trace will be maintained through this pattern (with an extra step, but that should be fine). If handlebars set any other properties we could just copy them over.

Reply:
Syntax error from handlebars have all those properties which show how the syntax was invalid. For example `err.hash.line` is the line the error was on, `err.hash.text` was the bad text it encountered, etc. Node.js errors coming through here have their standard `err.code` property, etc.

Reply:
The last time there was a release (in April) the feature was working. It seems like something has changed and it is no longer working for some reason. I'm trying to track down what is causing it to fail.

Reply:
Oh, haha, I forgot I was running handlebars 4.3.1 locally for testing. Yes, the feature still works fine on 4.0.14. So I'm trying to determine what changed in handlebars.

Reply:
It is something in the 4.2.1 release, as it works with handlebars 4.2.0 but does not work with 4.2.1

Reply:
Filed a bug in handlebars.js project: https://github.com/wycats/handlebars.js/issues/1562

Reply:
@dougwilson Good find.